test/per_field_misspellings_test.rb

```diff
@@ -0,0 +1,147 @@
+      misspellings: { fields: { name: false, color: false } }
+    }
+    assert_search "Sriracha", ["Sriracha"], params
+    assert_search "blue", ["Sriracha"], params
+  end
+
+  def test_misspelling_enabled
+    store [{name: "Sriracha", color: "blue"}]
+    assert_search "siracha", ["Sriracha"],
+      fields: [:name, :color],
+      misspellings: { fields: { name: true } }
+    assert_search "clue", ["Sriracha"],
+      fields: [:name, :color],
+      misspellings: { fields: { color: true } }
+  end
+
+  def test_misspelling_disabled
+    store [{name: "Sriracha", color: "blue"}]
+    assert_search "siracha", [],
+      fields: [:name, :color],
+      misspellings: { fields: { name: false } }
+    assert_search "clue", [],
+      fields: [:name, :color],
+      misspellings: { fields: { color: false } }
+  end
+
+  def test_misspelling_color
+    store [{name: "Sriracha", color: "blue"}]
+    assert_search "bluu", ["Sriracha"],
+      fields: [:name, :color],
+      misspellings: { fields: { name: false, color: true } }
+  end
+
+  def test_misspelling_multiple
+    store [
+      {name: "Greek Yogurt", color: "white"},
+      {name: "Green Onions", color: "yellow"}
+    ]
+    assert_search "greed", ["Greek Yogurt", "Green Onions"],
+      fields: [:name, :color],
+      misspellings: { fields: { name: true, color: false } }
+    assert_search "greed", [],
+      fields: [:name, :color],
+      misspellings: { fields: { name: false, color: true } }
+  end
+
+  def test_unspecified_field_uses_edit_distance_one
+    store [{name: "Bingo", color: "blue"}]
+    assert_search "bin", [],
+      fields: [:name, :color],
+      misspellings: { fields: { color: { edit_distance: 2 } } }
+    assert_search "bingooo", [],
+      fields: [:name, :color],
+      misspellings: { fields: { color: { edit_distance: 2 } } }
+    assert_search "mango", [],
+      fields: [:name, :color],
+      misspellings: { fields: { color: { edit_distance: 2 } } }
+    assert_search "bing", ["Bingo"],
+      fields: [:name, :color],
+      misspellings: { fields: { color: { edit_distance: 2 } } }
+  end
+
+  def test_field_uses_specified_edit_distance
+    store [{name: "Bingo", color: "yellow"}]
+    assert_search "yell", ["Bingo"],
+      fields: [:name, :color],
+      misspellings: { fields: { color: { edit_distance: 2 } } }
+    assert_search "yellowww", ["Bingo"],
+      fields: [:name, :color],
+      misspellings: { fields: { color: { edit_distance: 2 } } }
+    assert_search "yilliw", ["Bingo"],
+      fields: [:name, :color],
+      misspellings: { fields: { color: { edit_distance: 2 } } }
+  end
+
+  def test_misspelling_zucchini_transposition
+    store [{name: "zucchini", color: "green"}]
+    assert_search "zuccihni", [],
+      fields: [:name, :color],
+      misspellings: { fields: { name: {transpositions: false} } }
+    assert_search "grene", ["zucchini"],
+      fields: [:name, :color],
+      misspellings: { fields: { name: {transpositions: false} } }
```

Comment:
Would be good to have a helper method like `assert_misspellings` to DRY the tests. Also, let's add to existing misspellings test instead of a new file and use common prefix for method names, like `test_misspellings_field_...`.

Reply:
updated in https://github.com/ankane/searchkick/pull/1137/commits/28f1fb2da2fdc811351e80916e5b71f799cd21f6 (for `assert_misspellings`)
and https://github.com/ankane/searchkick/pull/1137/commits/29933aac41e6b0831830d371fd83a7658427a21c (combining the two files)